### PR TITLE
Feat/csv export

### PR DESCRIPTION
**Pull request by Scttpr**
Checklist

- [x] I've added a mention in CHANGELOG.md

### CODE REVIEW

File: pages/_campus/cars/index.vue [hunk 1]
```diff
@@ -4,6 +4,24 @@
...
+        <fa-icon
+          :icon="['fas', 'file-export']"
+          class="has-text-info"
+        />
+        Exporter CSV
+      </button>
+      <csv-modal
+        :csv-status="displayModal"
+        :pagination="pagination"
+        :api-call="$api.cars(campus.id, '*').getCars"
```
Comment (Nainterceptor) [inline pages/_campus/cars/index.vue]: Never use wildcard as mask

File: pages/_campus/phones/index.vue [hunk 2]
```diff
@@ -4,6 +4,23 @@
...
+        <fa-icon
+          :icon="['fas', 'file-export']"
+          class="has-text-info"
+        />
+        Exporter CSV
+      </button>
+      <csv-modal
+        :csv-status="displayModal"
+        :pagination="pagination"
+        :api-call="$api.phones(campus.id, '*').getPhones"
```
Comment (Nainterceptor) [inline pages/_campus/phones/index.vue]: Never use wildcard as mask

File: pages/_campus/pois/index.vue [hunk 3]
```diff
@@ -4,6 +4,24 @@
...
+        <fa-icon
+          :icon="['fas', 'file-export']"
+          class="has-text-info"
+        />
+        Exporter CSV
+      </button>
+      <csv-modal
+        :csv-status="displayModal"
+        :pagination="pagination"
+        :api-call="$api.pois(campus.id, '*').getPois"
```
Comment (Nainterceptor) [inline pages/_campus/pois/index.vue]: Never use wildcard as mask

File: pages/_campus/users/index.vue [hunk 4]
```diff
@@ -66,25 +88,36 @@
         />
       </div>
     </div>
+    <csv-modal
+      :csv-status="displayModal"
+      :pagination="modalType === 'users' ? usersPagination : driversPagination"
+      :api-call="modalType === 'users'
+        ? $api.campusUsers(campus.id, '*').getUsers
```
Comment (Nainterceptor) [inline pages/_campus/users/index.vue]: Never use wildcard as mask

File: pages/_campus/users/index.vue [hunk 5]
```diff
@@ -66,25 +88,36 @@
...
+        : $api.drivers(campus.id, '*').getDrivers"
```
Comment (Nainterceptor) [inline pages/_campus/users/index.vue]: Never use wildcard as mask

File: pages/categories/index.vue [hunk 6]
```diff
@@ -4,6 +4,24 @@
...
+        <fa-icon
+          :icon="['fas', 'file-export']"
+          class="has-text-info"
+        />
+        Exporter CSV
+      </button>
+      <csv-modal
+        :csv-status="displayModal"
+        :pagination="pagination"
+        :api-call="$api.categories('*').getCategories"
```
Comment (Nainterceptor) [inline pages/categories/index.vue]: Never use wildcard as mask

File: pages/pois/index.vue [hunk 7]
```diff
@@ -4,6 +4,24 @@
...
+        <fa-icon
+          :icon="['fas', 'file-export']"
+          class="has-text-info"
+        />
+        Exporter CSV
+      </button>
+      <csv-modal
+        :csv-status="displayModal"
+        :pagination="pagination"
+        :api-call="$api.pois(null, '*').getPois"
```
Comment (Nainterceptor) [inline pages/pois/index.vue]: Never use wildcard as mask

File: pages/ratings/index.vue [hunk 8]
```diff
@@ -31,23 +49,41 @@ export default {
...
   components: {
     ecList,
+    csvModal,
   },
   async asyncData({ $api, query }) {
     const offset = parseInt(query.offset, 10) || 0;
     const limit = parseInt(query.limit, 10) || 30;
-    const { data: ratings, pagination } = await $api.ratings('*').getRatings(offset, limit);
-    const ratingsToDisplay = ratings.map((rating) => ({
+    const { data, pagination } = await $api.ratings.getRatings('*', { offset, limit });
```
Comment (Nainterceptor) [inline pages/ratings/index.vue]: Never use wildcard as mask

File: components/csv-modal.vue [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+    'csv.mask': async function csvSeparator() {
+      await this.recalcDownloadLinks();
+    },
+  },
+  async created() {
+    this.csv = {
+      separator: ';',
+      delimiter: '"',
+      flatten: true,
+      mask: '*',
```
Comment (Nainterceptor) [inline components/csv-modal.vue]: Never use wildcard as mask

File: components/csv-modal.vue [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+      type: Function,
+      required: true,
+    },
+    hasMask: {
+      type: Boolean,
+      default: false,
+    },
+    fields: {
+      type: Array,
+      default: () => (['*']),
```
Comment (Nainterceptor) [inline components/csv-modal.vue]: Never use wildcard as mask, so don't put is as default

File: components/csv-modal.vue [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+        });
+    },
+    toggle(force) {
+      this.$emit('toggleModal', force);
+    },
+  },
+};
+</script>
+
+<style scoped>
```
Comment (Nainterceptor) [inline components/csv-modal.vue]: Empty

**Comment by Scttpr** [comment]
[Ajouter export CSV générique](https://trello.com/c/wz3t1Qor/155-ajouter-export-csv-g%C3%A9n%C3%A9rique)

File: pages/_campus/phones/index.vue [hunk 12]
```diff
@@ -66,17 +86,24 @@ export default {
...
+      .getPhones({ offset, limit });
     return {
       phones: data,
       pagination,
     };
   },
+  data() {
+    return {
+      displayModal: false,
+      mask: 'id,imei,model(label),number,owner(_id),campus,state,comments',
```
Comment (Nainterceptor) [inline pages/_campus/phones/index.vue]: It's better to user `id` and not `_id`, API do magics to rename the key :)

File: pages/_campus/cars/index.vue [hunk 13]
```diff
@@ -3,6 +3,12 @@
     <crud-header
       title="Véhicules"
       :to-create-new="campusLink('cars-new')"
+      upload-csv
```
Comment (Nainterceptor) [inline pages/_campus/cars/index.vue]: Just a naming thing, I think it's better to by symmetic, upload-csv / download-csv or import-csv / export-csv

**Pull request merged**